lib/Controller/FrontendController.php

```diff
@@ -79,6 +91,61 @@ public function __set($name, $value)
         throw new \RuntimeException(sprintf('Trying to set unknown property "%s"', $name));
     }
 
+    /**
+     * @required
+     * @param ResponseHeaderResolver $responseResolver
+     */
+    public function setResponseResolver(ResponseHeaderResolver $responseResolver): void
+    {
+        $this->responseResolver = $responseResolver;
+    }
+
+    /**
+     * @required
+     * @param RequestStack $requestStack
+     */
+    public function setRequestStack(RequestStack $requestStack): void
+    {
+        $this->requestStack = $requestStack;
+    }
+
+    /**
+     * @required
+     * @param DocumentResolver $documentResolver
+     */
+    public function setDocumentResolver(DocumentResolver $documentResolver): void
```

Comment:
```suggestion
    public function setDocumentResolver(DocumentResolver $documentResolver)
```
this can be a BC break if implemented in the subclass. e.g. https://github.com/pimcore/portal-engine/blob/2.x/src/Controller/Rest/Api/AbstractRestApiController.php#L149

Reply:
not sure if we really introduce these methods in abstract class